apps/prairielearn/src/pages/instructorAssessmentUploads/instructorAssessmentUploads.html.ts

```diff
@@ -0,0 +1,316 @@
+student2@example.com,1,addTwoNumbers,78.92,
+student2@example.com,1,matrixMultiply,100,Great job!</pre
+    >
+    <p>
+      The <code>instance</code> column indicates which assessment instance to modify. The total
+      scores for the assessments will be automatically recalculated. The feedback will be attached
+      to the most recent student submission. To change the per-question points, replace the
+      <code>score_perc</code> column above with a <code>points</code> column.
+    </p>
+    <p>
+      By default the updates above change the manual portion of the grades for the instance
+      question, in such a way that the sum of the new manual portion with the existing autograding
+      portion matches the total listed in the <code>points</code> or <code>score_perc</code> columns
+      above. It is, however, possible to replace these columns with <code>manual_points</code> and
+      <code>auto_points</code> (or <code>manual_score_perc</code> and <code>auto_score_perc</code>)
+      for more fine-tuned change of grade components.
+    </p>
+  `;
+}
+
+function CsvHelpAssessmentInstanceScores() {
+  return html`
+    <p>Upload a CSV file like this:</p>
+    <pre class="ml-4">
+uid,instance,score_perc
+student1@example.com,1,63.5
+student2@example.com,1,100</pre
+    >
+    <p>
+      The example above will change the total assessment percentage scores for
+      <code>student1@example.com</code> to 63.5% and for <code>student2@example.com</code> to 100%.
+      The <code>instance</code> column indicates which assessment instance to modify, and should be
+      <code>1</code> if there is only a single instance per student.
+    </p>
+    <p>
+      Alternatively, the total assessment points can be changed with a CSV containing a
+      <code>points</code> column, like:
+    </p>
+    <pre class="ml-4">
+uid,instance,points
+student1@example.com,1,120
+student2@example.com,1,130.27</pre
+    >
+    <p>For assessments using group work, use the <code>group_name</code> column instead:</p>
+    <pre class="ml-4">
+group_name,instance,score_perc
+myhappygroup,1,95
+greatgroup,1,85</pre
+    >
+  `;
+}
+
+function UploadInstanceQuestionScoresModal({ csrfToken }: { csrfToken: string }) {
+  return Modal({
+    id: 'upload-instance-question-scores-form',
+    title: 'Upload new question scores',
+    formEncType: 'multipart/form-data',
+    body: html`
+      ${CsvHelpInstanceQuestionScores()}
+      <div class="form-group">
+        <div class="custom-file">
+          <input
+            type="file"
+            name="file"
+            class="custom-file-input"
+            id="uploadInstanceQuestionScoresFileInput"
+          />
+          <label class="custom-file-label" for="uploadInstanceQuestionScoresFileInput">
+            Choose CSV file
+          </label>
+        </div>
+      </div>
+    `,
+    footer: html`
+      <div class="d-flex justify-content-end">
+        <div class="form-group mb-0">
```

Comment:
Are these two divs required, either here or in other modals below? We don't have these in any other modals.

Reply:
No, removed in [ca5cff9](https://github.com/PrairieLearn/PrairieLearn/pull/9777/commits/ca5cff947e437f62abd1fa09c931952a95344f31).